ts/screens/onboarding/OnboardingNotificationsPreferencesScreen.tsx

```diff
@@ -0,0 +1,158 @@
+import I18n from "../../i18n";
+import { profileRemindersPreferenceSelector } from "../../store/reducers/profile";
+import { useIODispatch } from "../../store/hooks";
+import { profileUpsert } from "../../store/actions/profile";
+import { ReminderStatusEnum } from "../../../definitions/backend/ReminderStatus";
+import { showToast } from "../../utils/showToast";
+import { IOStackNavigationProp } from "../../navigation/params/AppParamsList";
+import { OnboardingParamsList } from "../../navigation/params/OnboardingParamsList";
+import BaseScreenComponent from "../../components/screens/BaseScreenComponent";
+import { emptyContextualHelp } from "../../utils/emptyContextualHelp";
+import Switch from "../../components/ui/Switch";
+import FooterWithButtons from "../../components/ui/FooterWithButtons";
+import { BlockButtonProps } from "../../components/ui/BlockButtons";
+import { InfoBox } from "../../components/box/InfoBox";
+import { IOColors } from "../../components/core/variables/IOColors";
+import { H5 } from "../../components/core/typography/H5";
+import customVariables from "../../theme/variables";
+import { NotificationsPreferencesPreview } from "./components/NotificationsPreferencesPreview";
+
+const styles = StyleSheet.create({
+  separator: {
+    backgroundColor: customVariables.itemSeparator,
+    height: StyleSheet.hairlineWidth
+  },
+  bottomSpacer: {
+    marginBottom: customVariables.spacerLargeHeight
+  }
+});
+
+export type OnboardingNotificationsPreferencesScreenNavigationParams = {
+  isFirstOnboarding: boolean;
+};
+
+type Props = {
+  navigation: CompatNavigationProp<
+    IOStackNavigationProp<
+      OnboardingParamsList,
+      "ONBOARDING_NOTIFICATIONS_PREFERENCES"
+    >
+  >;
+};
+
+const continueButtonProps = (
+  isLoading: boolean,
+  onPress: () => void
+): BlockButtonProps => ({
+  block: true,
+  onPress,
+  title: I18n.t("onboarding.notifications.continue"),
+  isLoading
+});
+
+const loadingButtonProps = (): BlockButtonProps => ({
+  block: true,
+  onPress: undefined,
+  title: "",
+  disabled: true,
+  style: { backgroundColor: IOColors.greyLight, width: "100%" },
+  isLoading: true,
+  iconColor: IOColors.bluegreyDark
+});
+
+const OnboardingNotificationsPreferencesScreen = (props: Props) => {
+  const dispatch = useIODispatch();
+
+  const [remindersEnabled, setRemindersEnabled] = useState(true);
+  const [isUpserting, setIsUpserting] = useState(false);
+
+  const remindersPreference = useSelector(profileRemindersPreferenceSelector);
+
+  const isError = pot.isError(remindersPreference);
+  const isUpdating = pot.isUpdating(remindersPreference);
+
+  const isFirstOnboarding = props.navigation.getParam("isFirstOnboarding");
+
+  useEffect(() => {
+    if (isError && isUpserting) {
+      showToast(I18n.t("profile.preferences.notifications.error"));
+    }
+    if (!isUpdating) {
+      setIsUpserting(false);
+    }
+  }, [isError, isUpdating, isUpserting]);
```

Comment:
In case this was useful just to avoid the toast visualization when we come to the screen for the first time, may we handle this scenario with the `useOnFirstRender` hook?